### DISCUSSION TITLE
LaserGRBL workspace orientation does not match my machine

### DISCUSSION THREAD

**Discussion by Serv93** [arkypita/LaserGRBL#1813] [Q&A]
![Screenshot (2)](https://user-images.githubusercontent.com/103652891/163408608-5b7ebe86-7fc1-4d88-aac7-5a4ba56526da.png)
![Screenshot (3)](https://user-images.githubusercontent.com/103652891/163408645-725c381c-3df2-43de-959d-f9ac693aea27.png)
The homing position for my machine is top right rear. When I jog the machine moves in the correct direction. The laserGRBL machine position marker on screen shows the at the lower left corner. So essentially everything is running backwards.

**Comment by StuartB4** [comment]
You may need to change $3. Here's a table of the different numbers it should be set to to invert an axis or all axis.

![image](https://user-images.githubusercontent.com/16980026/163434010-450da6b3-93ca-45af-8056-93aaaba80514.png)

Also check $23
$23 - Homing dir invert, mask

By default, Grbl assumes your homing limit switches are in the positive direction, first moving the z-axis positive, then the x-y axes positive before trying to precisely locate machine zero by going back and forth slowly around the switch. If your machine has a limit switch in the negative direction, the homing direction mask can invert the axes' direction. It works just like the step port invert and direction port invert masks, where all you have to do is send the value in the table to indicate what axes you want to invert and search for in the opposite direction.

**Reply by Serv93** [reply]
Thanks for the help. It is much appreciated. I did set the $3 at 3. My machine is homed to the right rear but the target cross (blue at oo )in GRBL laser is in lower left corner. When I try to toggle, the machine runs in the correct direction but the target marker disappears into negative land. I have seen videos where all is set up like I have described above, but the target maker moves in the positive direction as it should when toggled.

Thanks
![Screenshot (4)](https://user-images.githubusercontent.com/103652891/163442339-2ceaacd6-bbe6-4c84-8d3d-701036931827.png)
